var/spack/repos/builtin/packages/percept/package.py

```diff
@@ -0,0 +1,75 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Percept(CMakePackage):
+    """Parallel mesh refinement and adaptivity tools for the finite
+    element method.
+    """
+
+    homepage = "https://github.com/PerceptTools/percept"
+    url      = "https://github.com/PerceptTools/percept.git"
+
+    # This package file was created at percept
+    # commit dc1c8ec0175213146ac139946beca185a84c22e8
+    version('master',
+            git='https://github.com/PerceptTools/percept.git', branch='master')
+
+    depends_on('googletest')
+    depends_on('opennurbs@percept')
+    depends_on('boost+graph+mpi')
+    depends_on('yaml-cpp+pic~shared@0.5.3:')
+    # Percept was initially tested against Trilinos 12.12.1
+    depends_on('trilinos~shared+exodus+tpetra+epetra+muelu+belos+ifpack2+amesos2+zoltan+stk+boost~superlu-dist+superlu+hdf5+zlib+pnetcdf+aztec+sacado~openmp+shards+intrepid+cgns@master,12.12.1:')
```

Comment:
why `~superlu-dist` and alike? I am quite cure that if `Perspect` works with `trilinos~superlu-dist` (i.e. it does not need it), then it must work with `+superlu-dist`. 

Generally, `package.py` is NOT a place to put user preferences, but rather the place to state requirements. 
User preferences should go to `packages.yaml` where you can add trilinos variant `~superlu-dist` to avoid compiling things you don't need/use if they are turned on by default in that given package. 

Please make this line reflect requirements of the package.

Reply:
Percept requires `+superlu` which conflicts with `superlu-dist`. This Trilinos line was also minimized during my work on Nalu and according to Percept's example Trilinos build script (which is based on Nalu's), this line is essentially the minimal set of requirements, except `openmp`, but I'm disabling it to be conservative here because it seems to cause problems.

Reply:
ah, ok, sorry, i mis-interpreted your intentions

Reply:
No problem. I had this exact same conversation with Adam Stewart 🙂